pkg/sentry/platform/kvm/machine_amd64.go

```diff
@@ -367,3 +368,41 @@ func rdonlyRegionsForSetMem() (phyRegions []physicalRegion) {
 func availableRegionsForSetMem() (phyRegions []physicalRegion) {
 	return physicalRegions
 }
+
+var execRegions []region
+
+func init() {
+	applyVirtualRegions(func(vr virtualRegion) {
+		if excludeVirtualRegion(vr) || vr.filename == "[vsyscall]" {
+			return
+		}
+
+		if vr.accessType.Execute {
+			execRegions = append(execRegions, vr.region)
+		}
+	})
+}
+
+func (m *machine) mapUpperHalf(pageTable *pagetables.PageTables) {
+	for _, r := range execRegions {
```

Comment:
I think it's probably clearer just to do this scan inline here. Not much more expensive (and saves having to keep the execRegions around forever). E.g.

execRegions := applyVirtualRegions(func(vr virtualRegion) {
  ...
})
for _, r := range execRegions {
  ...
}